nuxeo-retention-web/elements/nuxeo-retention-analytics.js

```diff
@@ -0,0 +1,351 @@
+            group-limit="10"
+            data="{{expiredRetention}}"
+            index="[[index]]"
+          >
+          </nuxeo-repository-data>
+
+          <nuxeo-card heading="[[i18n('retentionAnalytics.retentionStatus')]]">
+            <chart-pie
+              values="[[_valuesRetentionStatus(noRetention, futureRetention, expiredRetention)]]"
+              labels="[[_labelsRetentionStatus()]]"
+              options='{ "legend": { "display": true, "position": "bottom", "labels": { "boxWidth": 12 } }, "animation": false }'
+            >
+            </chart-pie>
+          </nuxeo-card>
+
+          <!-- LegalHold by Document Type -->
+          <nuxeo-repository-data
+            start-date="[[_formatDate(startDate)]]"
+            end-date="[[_extendEndDate(endDate)]]"
+            date-field="dc:created"
+            grouped-by="ecm:primaryType"
+            where='{"ecm:hasLegalHold": "true"}'
+            group-limit="20"
+            data="{{legalHoldByDocType}}"
+            index="[[index]]"
+          >
+          </nuxeo-repository-data>
+
+          <nuxeo-card heading="[[i18n('retentionAnalytics.legalHoldByDocType')]]">
+            <chart-pie
+              values="[[_values(legalHoldByDocType)]]"
+              labels="[[_labels(legalHoldByDocType)]]"
+              options='{ "legend": { "display": true, "position": "bottom", "labels": { "boxWidth": 12 } }, "animation": false }'
+            >
+            </chart-pie>
+          </nuxeo-card>
+        </div>
+
+        <!-- Expiration next 12 months -->
+        <nuxeo-repository-data
+          start-date="[[_formatDateNow()]]"
+          end-date="[[_formatDateFutureInMonths(12)]]"
+          with-date-intervals="month"
+          date-field="ecm:retainUntil"
+          data="{{expirationPerMonth}}"
+          index="[[index]]"
+        >
+        </nuxeo-repository-data>
+
+        <nuxeo-card class="chartBar" heading="[[i18n('retentionAnalytics.expirationNext12Months.heading')]]">
+          <chart-bar
+            labels="[[_futureExpirationLabels(expirationPerMonth)]]"
+            values="[[_values(expirationPerMonth)]]"
+            style="min-width:800px;"
+            options='{ "legend": { "display": false }, "animation": false }'
+          >
+          </chart-bar>
+        </nuxeo-card>
+      </nuxeo-page>
+    `;
+  }
+
+  static get is() {
+    return 'nuxeo-retention-analytics';
+  }
+
+  static get properties() {
+    return {
+      index: {
+        type: String,
+        value: '_all',
+      },
+      startDate: {
+        type: String,
+        value: moment().subtract(3, 'year').format('YYYY-MM-DD'),
+      },
+      endDate: {
+        type: String,
+        value: moment().format('YYYY-MM-DD'),
+      },
+    };
+  }
+
+  _values(data) {
+    return [
+      data.map((entry) => {
+        if (Array.isArray(entry.value)) {
+          return entry.value.map(this._values);
+        }
+        return entry.value;
+      }),
+    ];
```

Comment:
Did find new todo this things as part of ES6, as we spoke the last time.

Reply:
@RSalem07 , so in this code we are dealing w an array of objects. If we ignore the array check - we don't have any situation where that condition could appear AFAIK, we could simplify the code to:

`_values(data) {
    return [ data.map((entry) => entry.value) ];
  }`

WDYT?